community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/UpdateGraph.scala

```diff
@@ -78,20 +80,28 @@ case class UpdateGraph(mutatingPatterns: Seq[MutatingPattern] = Seq.empty) {
       )
   }
 
+  def setLabelOverlap(qg: QueryGraph): Boolean = {
```

Comment:
Can't you just add the labels from `SetLabelPattern` and add to the output of `labels` and let the `nodeOverlap` handle the overlap? Does it matter if label writes come from a `create` or `set`